catenets/datasets/dataset_ihdp.py

```diff
@@ -0,0 +1,268 @@
+    cate_true_out: pd.DataFrame of list
+        Average treatment effects for the testing set
+    """
+
+    X, y, w, mu0, mu1 = (
+        data_train["X"],
+        data_train["y"],
+        data_train["w"],
+        data_train["mu0"],
+        data_train["mu1"],
+    )
+
+    X_t, _, _, mu0_t, mu1_t = (
+        data_test["X"],
+        data_test["y"],
+        data_test["w"],
+        data_test["mu0"],
+        data_test["mu1"],
+    )
+    if setting == "D":
+        y[w == 1] = y[w == 1] + mu0[w == 1]
+        mu1 = mu0 + mu1
+        mu1_t = mu0_t + mu1_t
+
+    if rescale:
+        # rescale all outcomes to have similar scale of CATEs if sd_cate > 1
+        cate_in = mu0 - mu1
+        sd_cate = np.sqrt(cate_in.var())
+
+        if sd_cate > 1:
+            # training data
+            error = y - w * mu1 - (1 - w) * mu0
+            mu0 = mu0 / sd_cate
+            mu1 = mu1 / sd_cate
+            y = w * mu1 + (1 - w) * mu0 + error
+
+            # test data
+            mu0_t = mu0_t / sd_cate
+            mu1_t = mu1_t / sd_cate
+
+    cate_true_in = mu1 - mu0
+    cate_true_out = mu1_t - mu0_t
+
+    if return_pos:
+        return X, y, w, cate_true_in, X_t, cate_true_out, mu0, mu1, mu0_t, mu1_t
+
+    return X, y, w, cate_true_in, X_t, cate_true_out
+
+
+def get_one_data_set(D: dict, i_exp: int, get_po: bool = True) -> dict:
+    """
+    Helper for getting the IHDP data for one experiment. Adapted from https://github.com/clinicalml/cfrnet
+
+    Parameters
+    ----------
+    D: dict or pd.DataFrame
+        All the experiment
+    i_exp: int
+        Experiment number
+
+    Returns
+    -------
+    data: dict or pd.Dataframe
+        dict with the experiment
+    """
+    D_exp = {}
+    D_exp["X"] = D["X"][:, :, i_exp - 1]
+    D_exp["w"] = D["w"][:, i_exp - 1 : i_exp]
+    D_exp["y"] = D["y"][:, i_exp - 1 : i_exp]
+    if D["HAVE_TRUTH"]:
+        D_exp["ycf"] = D["ycf"][:, i_exp - 1 : i_exp]
+    else:
+        D_exp["ycf"] = None
+
+    if get_po:
+        D_exp["mu0"] = D["mu0"][:, i_exp - 1 : i_exp]
+        D_exp["mu1"] = D["mu1"][:, i_exp - 1 : i_exp]
+
+    return D_exp
+
+
+def load(data_path: Path, *args: Any, **kwargs: Any) -> Tuple:
+    """
+    Get IHDP train/test datasets with treatments and labels.
+
+    Parameters
+    ----------
+    data_path: Path
+        Path to the dataset csv. If the data is missing, it will be downloaded.
+
+
+    Returns
+    -------
+    X: pd.Dataframe or array
+        The training feature set
+    w: pd.DataFrame or array
+        Training treatment assignments.
+    y: pd.Dataframe or array
+        The training labels
+    training potential outcomes: pd.DataFrame or array.
+        Potential outcomes for the training set.
+    X_t: pd.DataFrame or array
+        The testing feature set
+    testing potential outcomes: pd.DataFrame of array
+        Potential outcomes for the testing set.
+    """
+    data_train, data_test = load_raw(data_path)
+
+    exp = 1
```

Comment:
This should be an argument; in IHDP experiments you want to report on all dataset indices